### PR TITLE
[Merged by Bors] - feat: the volume in a MeasureSpace over an empty index is the Dirac mass

### PR DESCRIPTION
**Pull request by sgouezel**
---

- [x] depends on: #8816 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Constructions/Pi.lean [hunk 1]
```diff
@@ -412,14 +412,22 @@ instance {α : ι → Type*} [∀ i, MeasureSpace (α i)] [∀ i, SigmaFinite (v
     SigmaFinite (volume : Measure (∀ i, α i)) :=
   pi.sigmaFinite _
 
-theorem pi_of_empty {α : Type*} [IsEmpty α] {β : α → Type*} {m : ∀ a, MeasurableSpace (β a)}
-    (μ : ∀ a : α, Measure (β a)) (x : ∀ a, β a := isEmptyElim) : Measure.pi μ = dirac x := by
+attribute [-instance] Fintype.ofIsEmpty in
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/MeasureTheory/Constructions/Pi.lean:415]: There's no diamond there:
```lean
import Mathlib

example : Fin.fintype 0 = Fintype.ofIsEmpty := rfl
```

[the remark this answers is not in the document]
Comment (sgouezel) [reply Mathlib/MeasureTheory/Constructions/Pi.lean:415]: #8816

[the remark this answers is not in the document]
Comment (alreadydone) [reply Mathlib/MeasureTheory/Constructions/Pi.lean:415]: I see. All computable Fintype instances on an empty type reduces to the empty list, so no diamond is possible and the instance Fintype.ofIsEmpty is safe in this sense. Maybe we can change the noncomputable instances to Finite instances, but Fintype is  still unavoidable for Finset.sum etv.

**Comment by RemyDegenne** [comment]
Thanks!
bors r+

**Pull request closed without merging**